### PR TITLE
A series of Public swagger update (modification) 

### PR DESCRIPTION
**Pull request by Yiliu-microsoft**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Dataset.json
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Dataset.json]: No point in changing the preview swagger as we are not releasing new SDK-s for this.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: This is not a good description

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: You can't delete an existing type

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (nschonni) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: addtional -> additional
```suggestion
        "additionalNullValues": {
```

Comment (wuchunchunt19) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: additionalNullValues is a private peroperty. Please remove it from public swagger. Thanks.

Comment (Yiliu-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: ok

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (Yiliu-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: ok

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (Yiliu-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: I asked the owner and he said that this modification is feasible

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Dataset.json
Comment (Yiliu-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Dataset.json]: ok

**Comment by hvermis** [comment]
@Yiliu-microsoft Please make sure to fix all checks as well

**Review by hvermis — CHANGES_REQUESTED** [review]
Please address comments

**Review by davidzhaoyue — CHANGES_REQUESTED** [review]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (davidzhaoyue) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: Where is ParquetSource?

**Review by davidzhaoyue — CHANGES_REQUESTED** [review]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (davidzhaoyue) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: This doesn't align with the latest private swagger.

**Review by davidzhaoyue — CHANGES_REQUESTED** [review]

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6429 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6429/commits/0d8ebe7c0f2c3fa3ce0672be21080b30e737c3a0

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (Yiliu-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: Hi David, I updated "ParquetSink"

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (wuchunchunt19) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: There is no treatEmptyAsNull in latest private swagger, please help remove it.

Comment (Yiliu-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: ok

**Review by wenbof-zz — COMMENTED** [review]
"SapTableLinkedService": {
      "x-ms-discriminator-value": "SapTable",
      "description": "SAP Table Linked Service.",

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json
Comment (wenbof) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json]: "$ref": "#/definitions/SapTableLinkedServiceTypeProperties"

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json
Comment (wenbof) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json]: "description": "SAP Table Linked Service.",

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json
Comment (wenbof) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json]: SapTable

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (wenbof) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: Sap Table Dataset part needs to be added as well

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (nschonni) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]:
```suggestion
          "description": "The maximum value of column specified in partitionColumnName that will be used for proceeding range partitioning. Type: string (or Expression with resultType string)."
```

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (nschonni) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]:
```suggestion
          "description": "The minimum value of column specified in partitionColumnName that will be used for proceeding range partitioning. Type: string (or Expression with resultType string)."
```

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (nschonni) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]:
```suggestion
          "description": "The maximum value of partitions the table will be split into. Type: integer (or Expression with resultType string)."
```

**Review by nschonni — COMMENTED** [review]
minor typos

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (shahabhijeet) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: breaking change in a stable version?
Any reason this cannot be done in the next API version?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (shahabhijeet) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: breaking change

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
I will need an approval from someone from your team in addition to addressing to couple of breaking changes you have introduced in this PR

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (Yiliu-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: Ok, I invited Hermine to explain this change, please wait a moment.

Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: @shahabhijeet the type is changing from string to object, so it's not API breaking change as our service will accept both.
We could mark the SDK-s with breaking tag if you think it's appropriate, however we had another PR like this that wasn't marked.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: same here

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: The description is longer in private repo, please copy from there. Also, there are two more description updates for schemaMappings and columnMapping property, where we advise the customers not to use them and use this instead. Please sync with David.

**Review by hvermis — APPROVED** [review]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (shahabhijeet) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: @hvermis it is a binary breaking change.
I was concerned about a scenario where your customer has written code against your model as
```
JsonFormat.FilePattern.ToLower().StartWith("array")
```
Also what I noticed that you have modeled your enum as "modelAsString", but you would like to make a change from string to object?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
I have asked ARM review for this.
What I would like to know is, can you make such changes in stable APIs?
I would like to understand such process, which will help me in future reviews.

**Comment by hvermis** [comment]
@shahabhijeet All these changes are again to support oneOf scenario. The reason we are changing types to object is so our customers can parametrize those properties. Our UI and backend already allow this, only SDK-s are pending.

**Comment by hvermis** [comment]
@shahabhijeet and @KrisBash Could you please help get this PR in? It contains a lot of features that our customers are waiting for and we need to release our SDKs.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Pipeline.json
Comment (KrisBash) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Pipeline.json]: If all allowed enum values are a string, why change to object?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Pipeline.json
Comment (KrisBash) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Pipeline.json]: Like the above, does this need to be an object if allowed values are all strings?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json
Comment (KrisBash) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json]: Friendly reminder: please ensure that the server never returns secrets on get nor logs requests.

**Review by KrisBash — CHANGES_REQUESTED** [review]
Discussion about SDK breaking changes aside, there were quite a few properties changed to type: object, but with string enums and modelAsString=true. That seems spurious?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Pipeline.json
Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Pipeline.json]: This is to support Expressions on these properties. OpenApi 2.0 doesn't support oneOf, and we are forced to use object type in order to support parameterization, which is an important feature of ADF for our customers. It allows them to treat their entities  as templates and pass different values for properties.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json
Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json]: This is already taken care of - we encrypt all secrets, and our service returns only encrypted values.

**Comment by hvermis** [comment]
@KrisBash This is the PR that we need merged. Can you please finish reviewing this one? I will follow up with Yi to close any others, if they are irrelevant.

**Comment by nschonni** [comment]
FYI, you can add `bzip` to https://github.com/Azure/azure-rest-api-specs/blob/master/custom-words.txt if you want the spellcheck to pass. The other issue it was flagging in a different PR was fixed in master

**Comment by hvermis** [comment]
@Yiliu-microsoft  Can you please follow up on Nick's suggestion and add bzip to the file he mentioned in this PR? Let's get our build clean.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: @shahabhijeet We'll mark the SDK release as breaking.

**Comment by hvermis** [comment]
@KrisBash @shahabhijeet  We need to release this new SDK with Powershell release that is going out on 6/18.
Can you please help as we have a lot of customers waiting for the features in this PR?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json
Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json]: @Yiliu-microsoft  Please make sure to fix this before merge. It is interesting that tools didn't catch this. @shahabhijeet there was already another type with the same x-ms-discriminator-value, the verification builds on swagger don't catch it. Is it expected?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: Why are the type names here Setting vs Settings?

Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: Applies to all names

Comment (Yiliu-microsoft) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: I change "description": "Connector read settings."  to  "description": "Connector read setting."

**Review by KrisBash — APPROVED** [review]
No objection from ARM, assuming changes are legitimate in SDK

**Pull request merged**
